### ISSUE TITLE
Fails to download anime that is only on streamsb

### ISSUE THREAD

**Issue by Dabbing-Guy** [sdaqo/anipy-cli#63]
**Describe the bug**
This bug happens when an anime episode on gogo is not actually on the gogo server and needs to be streamed from streamsb. It causes the program to crash when trying to watch it or download it.

You can also tell that it needs to be streamed from streamsb because that is what happens on gogo.

Link to episode on gogo: https://gogoanime.gg/cowboy-bebop-dub-episode-15

**To Reproduce**
Try watching or downloading Cowboy Bebop (Dub) episode 15.

**Expected behavior**
Normal Operation. 

**Info**
```
Name: anipy-cli
Version: 2.4.18
Summary: Little tool in python to watch anime from the terminal (the better way to watch anime)
Home-page: https://github.com/sdaqo/anipy-cli
Author: sdaqo
Author-email: sdaqo.dev@protonmail.com
License: GPL-3.0
Location: /home/michael/.local/lib/python3.10/site-packages
Requires: better-ffmpeg-progress, bs4, kitsu.py-extended, m3u8, moviepy, pycryptodomex, pypresence, python-dateutil, requests, tqdm
Required-by:
```

**Platform**
 - OS: Bedrock Linux
- Python version 3.10.4

**Additional context**
Output (from both trying to download and watch normally):

```
❯ anipy-cli -d
***Download Mode***
Downloads are stored in: /home/michael/anime
Search Kitsu for anime in Season? (y|n):
>>
Search: cowboy bebop
Add another search: (y|n)


Current:  cowboy bebop
[1] Cowboy Bebop
[2] Cowboy Bebop (Dub)
[3] Cowboy Bebop: Tengoku no Tobira
[4] Cowboy Bebop: Tengoku no Tobira (Dub)
[5] Cowboy Bebop Knockin` on Heaven`s Door
Enter Number: 2
Episode [1-26]: 15
anipy-cli: error: quality not avalible, using default
--------------------
Downloading: Cowboy Bebop (Dub) EP: 15 - hlsp
Type: m3u8
Downloader: internal
Traceback (most recent call last):
  File "/home/michael/.local/bin/anipy-cli", line 8, in <module>
    sys.exit(main())
  File "/home/michael/.local/lib/python3.10/site-packages/anipy_cli/run_anipy_cli.py", line 8, in main
    cli.main()
  File "/home/michael/.local/lib/python3.10/site-packages/anipy_cli/cli.py", line 641, in main
    download_cli(args.quality, args.ffmpeg, args.no_kitsu)
  File "/home/michael/.local/lib/python3.10/site-packages/anipy_cli/cli.py", line 106, in download_cli
    download(show_entry, ffmpeg).download()
  File "/home/michael/.local/lib/python3.10/site-packages/anipy_cli/download.py", line 73, in download
    self.multithread_m3u8_dl()
  File "/home/michael/.local/lib/python3.10/site-packages/anipy_cli/download.py", line 256, in multithread_m3u8_dl
    self._m3u8_content = self._download_m3u8(
  File "/home/michael/.local/lib/python3.10/site-packages/anipy_cli/download.py", line 293, in _download_m3u8
    resp.raise_for_status()
  File "/home/michael/.local/lib/python3.10/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://www11.gogocdn.stream/videos/hls/fHLZTWB-TVC_A6YXgjehJg/1655596029/106825/72c2d7ea52bff63114544ca2a4b17e6d/ep.15.1648324057.m3u8

~/.local/lib/python3.10/site-packages/anipy_cli   20s
❯ anipy-cli
Search: cowboy bebop
[1] Cowboy Bebop
[2] Cowboy Bebop (Dub)
[3] Cowboy Bebop: Tengoku no Tobira
[4] Cowboy Bebop: Tengoku no Tobira (Dub)
[5] Cowboy Bebop Knockin` on Heaven`s Door
Enter Number: 2
Episode [1-26]: 15
anipy-cli: error: quality not avalible, using default
Traceback (most recent call last):
  File "/home/michael/.local/bin/anipy-cli", line 8, in <module>
    sys.exit(main())
  File "/home/michael/.local/lib/python3.10/site-packages/anipy_cli/run_anipy_cli.py", line 8, in main
    cli.main()
  File "/home/michael/.local/lib/python3.10/site-packages/anipy_cli/cli.py", line 659, in main
    default_cli(args.quality, player)
  File "/home/michael/.local/lib/python3.10/site-packages/anipy_cli/cli.py", line 43, in default_cli
    sub_proc = start_player(show_entry, rpc_client, player)
NameError: name 'rpc_client' is not defined
```

**Comment by sdaqo** [comment]
Support for a new stream provider will be somewhat a big thing. I will postpone that for now since looks like that isn't really on a lot of eps and we still have a PR open. This is on the wish-list for now. I will leave this open though.

**Comment by HimanshuChaudhary17** [comment]
After studying the issue, I propose that it be resolved by adding a check to see whether the episode requires streaming from streamsb or not. If that is the case, the programme should be able to handle the streamsb stream. This would keep the programme from crashing and ensure that it ran normally. This will be your temporary solution to the bug until you receive support from the new stream provider.

**Comment by sdaqo** [comment]
Since this issue is currently irrelevant  (the episode in the issue cowboy bebop  15 is not broken anymore and we dont have anything to test now anyways) I will close this issue for now and convert it to a discussion.  Support for multiple streaming providers is an idea I do have in mind but am currently to lazy to do. 

As for your idea @HimanshuChaudhary17 sure but I would just leave the option to retry/skip and just put a try/except around the call to download (in download_cli.py) so that users who download a big batch of anime and leave their PC unattended don't get stuck on the choice between retry/skip. Do go ahead and make a PR for this!
